File: drills/elixir/solutions/nc79_kth_largest_array.exs
defmodule Solution do
  # Sorting answers every k at once, which is more than asked for but is the
  # version nobody gets wrong. O(n log n).
  def find_kth_largest(nums, k) when k < 1 or k > length(nums), do: nil

  def find_kth_largest(nums, k), do: nums |> Enum.sort(:desc) |> Enum.at(k - 1)
end

File: drills/ts/solutions/nc79_kth_largest_array.ts
export function findKthLargest(nums: number[], k: number): number | null {
  // Sorting answers every k at once, which is more than asked for but is the
  // version nobody gets wrong. O(n log n).
  if (k < 1 || k > nums.length) return null;
  return [...nums].sort((a, b) => b - a)[k - 1];
}
